### PR TITLE
Dote dev

### PR DESCRIPTION
**Pull request by JDTobin**
Add DOT&E changes to XDS dev

### CODE REVIEW

File: app/es_api/utils/queries.py [hunk 1]
```diff
@@ -211,6 +213,45 @@ def more_like_this(self, doc_id):
...
+        uiConfig = configuration.xdsuiconfiguration
+        search_filters = SearchFilter.objects.filter(
+            xds_ui_configuration=uiConfig, active=True)
+
+        # create aggregations for each filter
+        self.add_search_aggregations(filter_set=search_filters)
+
+        # Skipping first response found and getting next three
+        # if len(self.search > 3):
+        self.search = self.search[1:4]
```
Comment (JDTobin) [inline app/es_api/utils/queries.py:247]: @mdonlon why are you skipping the first? also probably don't need filters or aggregation

File: app/es_api/views.py [hunk 2]
```diff
@@ -239,6 +240,49 @@ def get(self, request, doc_id):
             return HttpResponse(results, content_type="application/json")
 
 
+class GetSimilarCoursesView(APIView):
+    """This method defines an API for fetching results using the
+            more_like_this feature from elasticsearch for similar courses. """
```
Comment (JDTobin) [inline app/es_api/views.py:245]: @mdonlon this explicitly does not use more_like_this, it uses a multi_match

File: app/configurations/models.py [hunk 3]
```diff
@@ -198,7 +198,7 @@ class CourseInformationMapping(TimeStampedModel):
                                                   " elasticsearch")
 
     course_derived_from = models.CharField(max_length=200,
-                                           default="P2881_Core.DerivedFrom",
+                                           default="P2881-Core.DerivedFrom",
```
Comment (JDTobin) [inline app/configurations/models.py:201]: @mdonlon don't bother changing it back, but you don't typically need to change the default, as it is just the pre-populated value for the field when you go to create an instance of it

File: app/es_api/utils/queries.py [hunk 4]
```diff
@@ -3,7 +3,8 @@
 import logging
 
 from configurations.models import CourseInformationMapping, XDSConfiguration
-from core.models import CourseSpotlight, SearchFilter, SearchSortOption
+from core.models import CourseSpotlight, SearchFilter
+from core.models import SearchSortOption, SearchField
```
Comment (JDTobin) [inline app/es_api/utils/queries.py:7]: @mdonlon I am surprised lint didn't catch this, but these should be part of the same import statement, even if you have to split it across lines.  see `core/admin.py` if you need an example

File: app/configurations/models.py (hunk 3 above)
Comment (mdonlon) [reply app/configurations/models.py:201]: Noted

File: app/es_api/utils/queries.py (hunk 4 above)
Comment (mdonlon) [reply app/es_api/utils/queries.py:7]: lint said that the line was too long so I split it, I fixed this now and will push an update.

File: app/es_api/views.py (hunk 2 above)
Comment (mdonlon) [reply app/es_api/views.py:245]: I forgot to fix this comment, will edit and push fix.

File: app/es_api/utils/queries.py (hunk 1 above)
Comment (JDTobin) [reply app/es_api/utils/queries.py:247]: you should probably keep the user_organization_filtering in.  
what is the value of keyword coming in?  
if it is a UUID then you can either add that to the search query as not equal, or get 4 items returned and if one matches the UUID remove it from the list.  if keyword is a UUID you would also need to update your query.
if it is a competency or subject, then you could consider including the UUID of the object you are basing it off of as a parameter so you can add it to the search, or just send 4 responses to the UI and let the UI filter out any.

File: app/es_api/views.py (hunk 2 above)
Comment (JDTobin) [reply app/es_api/views.py:245]: still missing the update to this

File: app/es_api/utils/queries.py [hunk 5]
```diff
@@ -211,6 +213,45 @@ def more_like_this(self, doc_id):
 
         return response
 
+    def similar_courses(self, keyword=""):
+        """This method takes in a keyword and queries the elasticsearch index
+           for 3 courses with similar competnencies or subjects"""
```
Comment (JDTobin) [inline app/es_api/utils/queries.py:218]: @mdonlon spelling

File: app/es_api/utils/queries.py (hunk 1 above)
Comment (mdonlon) [reply app/es_api/utils/queries.py:247]: I'm sending what is being mapped into the competency field (currently KSA Alignment for DOT&E courses). I think the easiest way to is to send back 4 and have the UI filter out one if the UUID matches the UUID of what we used to send out the query in the first place. This query also might not be getting used in our Portal code, or having the same course might not be as big of an issue going forward.

**Pull request merged**